cmd/podman/create.go

```diff
@@ -785,11 +786,15 @@ func joinOrCreateRootlessUserNamespace(createConfig *cc.CreateConfig, runtime *l
 			if s != libpod.ContainerStateRunning && s != libpod.ContainerStatePaused {
 				continue
 			}
-			pid, err := prevCtr.PID()
+			data, err := ioutil.ReadFile(prevCtr.Config().ConmonPidFile)
```

Comment:
Do we need to do this with pods in root world as well.  If I run a pod in a UserNS Then the containers need to run in the same.

Reply:
I'll need to verify that.  Probably if we are using user namespaces we will need to join the parent username space first.

Reply:
@rhatdan @giuseppe Is it sufficient to ask runc to join the appropriate userns? We already have machinery for that built in, and I'd hope that runc knows the right order to join namespaces in.

Reply:
Yes I think that is all that is necessary.

Bottom line pods and containers need to all run in the same User Namespace, especially if they are sharing volumes.  
Otherwise stuff like the shared ipc or pid namespaces will get all screwed up.